Deltalake 0.7.0 with s3 feature compliation error due to rusoto_dynamodb version conflict

# Environment

**Delta-rs**: 0.7.0

**Binding**: Rust

**Environment**:
- **Cloud provider**: AWS
- **OS**: MacOS Venture
- **Other**:  rustc 1.67.1

***
# Bug

**What happened**:
It sounds like duplicate of #973 which is declared fixed but seems not yet. The version numbers of the conflict are different this time.

Trying to compile a simple program that enables the s3 feature fails to compile due to a version mismatch in rusoto_dynamodb dependency

**What you expected to happen**:
Successful compliation

**How to reproduce it**:
main.rs:
```rust
#[tokio::main]
async fn main() {
    let table = deltalake::open_table_with_version("s3://foo", 0).await?;
    let files = table.get_files();
    println!("files = {:?}", files);
}
```

cargo.toml
```toml
[package]
name = "rsuty-delta-aws"
version = "0.1.0"
edition = "2021"

[dependencies]
deltalake = { version = "0.7.0", features = ["s3"] }
tokio = { version = "1.25.0", features = ["full"] }

```

Then compilation using `cargo build` produces the following:

```
error[E0308]: mismatched types
    --> /Users/yousry/.cargo/registry/src/github.com-1ecc6299db9ec823/deltalake-0.7.0/src/storage/s3.rs:527:17
     |
526  |             let lock_client = dynamodb_lock::DynamoDbLockClient::new(
     |                               -------------------------------------- arguments to this function are incorrect
527  |                 dynamodb_client,
     |                 ^^^^^^^^^^^^^^^ expected struct `rusoto_dynamodb::generated::DynamoDbClient`, found struct `DynamoDbClient`
     |
     = note: struct `DynamoDbClient` and struct `rusoto_dynamodb::generated::DynamoDbClient` have similar names, but are actually distinct types
note: struct `DynamoDbClient` is defined in crate `rusoto_dynamodb`
    --> /Users/yousry/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_dynamodb-0.47.0/src/generated.rs:6449:1
     |
6449 | pub struct DynamoDbClient {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^
note: struct `rusoto_dynamodb::generated::DynamoDbClient` is defined in crate `rusoto_dynamodb`
    --> /Users/yousry/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_dynamodb-0.48.0/src/generated.rs:6449:1
     |
6449 | pub struct DynamoDbClient {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^
     = note: perhaps two different versions of crate `rusoto_dynamodb` are being used?
note: associated function defined here
    --> /Users/yousry/.cargo/registry/src/github.com-1ecc6299db9ec823/dynamodb_lock-0.4.2/src/lib.rs:341:12
     |
341  |     pub fn new(client: DynamoDbClient, opts: DynamoDbOptions) -> Self {
     |            ^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `deltalake` due to previous error
warning: build failed, waiting for other jobs to finish...
```


**More details**:
`cargo tree` can be used to track the issue but here are some screenshots.

`deltalake` itself references `rusoto_dynamodb` v 0.47.0
 
<img width="826" alt="image" src="https://user-images.githubusercontent.com/2197158/221984268-0b2e883d-d298-4f7c-ad17-752e9ad7221c.png">

`dynamodb_lock` references `rusoto_dynamodb` v 0.48.0

<img width="840" alt="image" src="https://user-images.githubusercontent.com/2197158/221984666-4cbf7ef1-cb0d-4726-a93f-fbdd226b9f42.png">


For some reason, it seems that `dynamodb_lock` has been released while still using `rusoto_dynamodb` v 0.48.0.

<img width="869" alt="image" src="https://user-images.githubusercontent.com/2197158/221986049-614d34c4-60fa-43bb-8025-41de2e0f921c.png">

I just wanted to share some references here and there although I am not really expert or across all those versioning changes.

--- a/dynamodb_lock/Cargo.toml
+++ b/dynamodb_lock/Cargo.toml
@@ -16,8 +16,8 @@ uuid = { version = "1.2", features = ["serde", "v4"] }
 
 # Waiting on fix for regression in 0.48:
 # https://github.com/rusoto/rusoto/issues/1980
-rusoto_core = { version = ">=0.47", default-features = false }
-rusoto_dynamodb = { version = ">=0.47", default-features = false }
+rusoto_core = { version = "0.47", default-features = false }
+rusoto_dynamodb = { version = "0.47", default-features = false }
 maplit = "1"
 tokio = { version = "1", features = ["fs", "macros", "rt", "io-util"] }
 

--- a/dynamodb_lock/Cargo.toml
+++ b/dynamodb_lock/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 name = "dynamodb_lock"
 description = "Distributed lock backed by Dynamodb"
-version = "0.4.2"
+version = "0.4.3"
 authors = ["Mykhailo Osypov <mykhailo.osypov@gmail.com>"]
 edition = "2018"
 readme = "README.md"
@@ -16,8 +16,8 @@ uuid = { version = "1.2", features = ["serde", "v4"] }
 
 # Waiting on fix for regression in 0.48:
 # https://github.com/rusoto/rusoto/issues/1980
-rusoto_core = { version = "0.47", default-features = false }
-rusoto_dynamodb = { version = "0.47", default-features = false }
+rusoto_core = { version = ">=0.47", default-features = false }
+rusoto_dynamodb = { version = ">=0.47", default-features = false }
 maplit = "1"
 tokio = { version = "1", features = ["fs", "macros", "rt", "io-util"] }
 

